### PR TITLE
change servicediscovery pkg to nameresolution pkg and remove mdns code

### PR DESCRIPTION
**Pull request by youngbupark**
# Description

- Change servicediscovery pkg to nameresolution pkg
- Remove mDNS dependency from dapr/dapr

## Issue reference

#916 

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [ ] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/components/nameresolution/registry.go [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package nameresolution
+
+import (
+	"fmt"
+
+	"github.com/dapr/components-contrib/nameresolution"
```
Comment (LMWF) [inline pkg/components/nameresolution/registry.go:11]: This appears 2x.

File: pkg/runtime/options.go [hunk 2]
```diff
@@ -4,23 +4,23 @@ import (
...
-		exporters        []exporters.Exporter
-		serviceDiscovery []servicediscovery.ServiceDiscovery
-		inputBindings    []bindings.InputBinding
-		outputBindings   []bindings.OutputBinding
-		httpMiddleware   []http.Middleware
+		secretStores    []secretstores.SecretStore
+		states          []state.State
+		pubsubs         []pubsub.PubSub
+		exporters       []exporters.Exporter
+		nameResolutions []nameresolution.NameResolution
```
Comment (LMWF) [inline pkg/runtime/options.go:20]: nameResolvers?

Comment (youngbupark) [reply pkg/runtime/options.go:20]: I would follow the existing convention to be consistent with the existing ones - `[component type]s`

**Pull request merged**